### PR TITLE
Introduce the new `cluster` option

### PR DESCRIPTION
**Pull request by topliceanu**
- Add tests to check the new cluster option.
- Implement support for the cluster option in the constructor.
- Bump up the library minor version #90

### CODE REVIEW

**Comment by topliceanu** [comment]
The moral of this PR: `configure you editor to remove trailing white spaces on save!`

File: lib/Pusher.php [hunk 1]
```diff
@@ -79,6 +79,7 @@ class Pusher
 	 * 	port - the http port
 	 * 	timeout - the http timeout
 	 * 	encrypted - quick option to use scheme of https and port 443.
+   * 	cluster - cluster name to connect to.
```
Comment (hamchapman) [inline lib/Pusher.php]: Whitespace at start of the line looks a bit off here

File: lib/Pusher.php [hunk 2]
```diff
@@ -140,6 +141,12 @@ public function __construct( $auth_key, $secret, $app_id, $options = array(), $h
 			}
 		}
 
+    // handle the case when 'host' and 'cluster' are specified in the options.
```
Comment (hamchapman) [inline lib/Pusher.php]: Same with this block, the whitespace looks a bit off

File: test/unit/pusherConstructorTest.php [hunk 3]
```diff
@@ -78,22 +78,53 @@ public function testEncryptedOptionWillBeOverwrittenByHostAndPortOptionsSetHostA
...
 			$pusher = new Pusher( 'app_key', 'app_secret', 'app_id', $options );
-			
+
 			$settings = $pusher->getSettings();
 			$this->assertEquals( 'http', $settings[ 'scheme' ] );
 			$this->assertEquals( 'test.com', $settings[ 'host' ] );
-		}		
+		}
+
+    public function testClusterSetsANewHost() {
```
Comment (hamchapman) [inline test/unit/pusherConstructorTest.php]: And here (sorry to be picky :P ) 

Looks like the spacing is a bit random throughout the project, but we should try and stay consistent with what's there already :)

Comment (tanuck) [inline test/unit/pusherConstructorTest.php]: You could include a linter in the travis build. The `new-lib` branch already does this, you could use that as an example.

Comment (mdpye) [inline test/unit/pusherConstructorTest.php]: The project appears to use tabs exclusively, but this PR adds a lot of spaces.

**Comment by topliceanu** [comment]
@hamchapman I reverted to using tabs instead of spaces. We should really change to using whitespaces.

@tanuck I agree that automated linting would be great to have. However, It's not the scope of this PR. I suggest creating a separate PR __just with that__ and I'll be more than happy to merge it.

File: test/unit/pusherConstructorTest.php (hunk 3 above)
Comment (topliceanu) [inline test/unit/pusherConstructorTest.php]: This project has a lot of trailing white spaces. My editor is wired to
automatically remove them. That is why this PR is so "busy". If needs be I
can make another PR ?!

On Wed, Feb 17, 2016 at 4:39 PM, Mike Pye <notifications@github.com> wrote:

> In test/unit/pusherConstructorTest.php
> <https://github.com/pusher/pusher-http-php/pull/91#discussion_r53192016>:
>
> >  			$settings = $pusher->getSettings();
> >  			$this->assertEquals( 'http', $settings[ 'scheme' ] );
> >  			$this->assertEquals( 'test.com', $settings[ 'host' ] );
> > -		}		
> > +		}
> > +
> > +    public function testClusterSetsANewHost() {
>
> The project appears to use tabs exclusively, but this PR adds a lot of
> spaces.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pusher/pusher-http-php/pull/91/files#r53192016>.
>



-- 
Alexandru Topliceanu
